src/http/one/TeChunkedParser.cc

```diff
@@ -115,55 +117,88 @@ Http::One::TeChunkedParser::parseChunkSize(Http1::Tokenizer &tok)
-
-            if (!skipKnown) {
-                if (ext.cmp("use-original-body",17) == 0 && tok.int64(useOriginBody, 10)) {
-                    debugs(94, 3, "Found chunk extension " << ext << "=" << useOriginBody);
-                    buf_ = tok.remaining(); // parse checkpoint
-                    continue;
-                }
-            }
-
-            debugs(94, 5, "skipping unknown chunk extension " << ext);
-
-            // unknown might have a value token or quoted-string
-            if (tok.quotedStringOrToken(value) && !tok.atEnd()) {
-                buf_ = tok.remaining(); // parse checkpoint
-                continue;
-            }
-
-            // otherwise need more data OR corrupt syntax
-            break;
-        }
-
-        if (!tok.atEnd())
-            buf_ = tok.remaining(); // parse checkpoint (unless there might be more token name)
+    while (parseOneChunkExtension(tok)) {
+        buf_ = tok.remaining(); // got one extension; there may be more
     }
 
-    if (skipLineTerminator(tok)) {
-        buf_ = tok.remaining(); // checkpoint
-        // non-0 chunk means data, 0-size means optional Trailer follows
+    if (skipLineTerminatorIfAny(tok)) {
+        buf_ = tok.remaining();
         parsingStage_ = theChunkSize ? Http1::HTTP_PARSE_CHUNK : Http1::HTTP_PARSE_MIME;
+        return true; // reached the end of extensions (if any)
+    }
+
+    static const CharacterSet NonLF = (CharacterSet::LF).complement().rename("non-LF");
+    if (tok.skipAll(NonLF) && tok.skip('\n'))
+        throw TexcHere("cannot parse chunk extension"); // <garbage> CR*LF
+    return false; // need more data
+}
+
+bool
+Http::One::CustomExtensionsParser::parseIntExtension(Parser::Tokenizer &tok, const SBuf &name, int64_t &value)
+{
+    assert(!tok.atEnd()); // guaranteed by ParseBws() after "="
+
+    const auto savedTok = tok;
+    int64_t parsedValue = 0;
+
+    if (!tok.int64(parsedValue, 10))
+        throw TexcHere(ToSBuf("invalid value for the decimal integer chunk extension ", name));
+
+    if (!tok.atEnd()) {
+        value = parsedValue;
         return true;
     }
 
+    tok = savedTok;
     return false;
 }
 
 bool
-Http::One::TeChunkedParser::parseChunkBody(Http1::Tokenizer &tok)
+Http::One::TeChunkedParser::parseOneChunkExtension(Tokenizer &tok)
+{
+    const auto savedTok = tok;
+    SBuf extName;
+
+    if (!ParseBws(tok) || // Bug 4492: IBM_HTTP_Server sends SP after chunk-size
+        !tok.skip(';') ||
+        !ParseBws(tok) || // Bg 4492: ICAP servers send SP before chunk-ext-name
+        !tok.prefix(extName, CharacterSet::TCHAR) ||
+        !ParseBws(tok)) {
+        tok = savedTok;
+        return false;
+    }
+
+    if (!tok.skip('='))
+        return true; // parsed a valueless chunk-ext
+
+    if (!ParseBws(tok)) {
+        tok = savedTok;
+        return false;
+    }
+
+    if (customExtensionsParser && customExtensionsParser->knownExtension(extName))
```

Comment:
```suggestion
    if (customExtensionsParser && customExtensionsParser->knownExtension(extName)) {
```

Please apply automated code formatting if you have not already.